connectors/byoc.py

```diff
@@ -116,32 +116,69 @@ async def preflight(self):
             indices=[CONNECTORS_INDEX, JOBS_INDEX], pipelines=[PIPELINE]
         )
 
-    async def get_list(self):
+    async def get_connectors(self, native_service_types=[], connectors_ids=[]):
         await self.client.indices.refresh(index=CONNECTORS_INDEX)
```

Comment:
Will not passing `native_service_types` or `connectors_ids` result in 0 results returned in this case?

Reply:
They can be empty because they are configured in yaml. 

I added default values to allow function calls like
```python
self.connectors.get_connectors(native_service_types=native_service_types)
self.connectors.get_connectors(connectors_ids=connectors_ids)
```

But I will add this at the beginning of the function:
```python
if len(native_service_types) == 0 and len(connectors_ids) == 0:
    return
```

I should also change the query to make it work for scenarios when `native_service_types` is empty or `connectors_ids` is empty